src/org/thoughtcrime/securesms/WebxdcActivity.java

```diff
@@ -122,26 +122,16 @@ protected void onCreate(Bundle state, boolean ready) {
     webSettings.setJavaScriptEnabled(true);
     webSettings.setAllowFileAccess(false);
     webSettings.setBlockNetworkLoads(!internetAccess);
-    webView.setNetworkAvailable(internetAccess);
     webSettings.setAllowContentAccess(false);
     webSettings.setGeolocationEnabled(false);
     webSettings.setAllowFileAccessFromFileURLs(false);
     webSettings.setAllowUniversalAccessFromFileURLs(false);
     webSettings.setDatabaseEnabled(true);
     webSettings.setDomStorageEnabled(true);
+    webView.setNetworkAvailable(internetAccess); // this does not block network but sets `window.navigator.isOnline` in js land
     webView.addJavascriptInterface(new InternalJSApi(), "InternalJSApi");
 
-
-    InputStream bootstrap_stream = getResources().openRawResource(R.raw.webxdc_wrapper);
-    byte[] bootstrap_bytes = new byte[0];
-    try {
-      bootstrap_bytes = new byte[bootstrap_stream.available()];
-      bootstrap_stream.read(bootstrap_bytes);
-      String bootstrapHtml = new String(bootstrap_bytes);
-      webView.loadDataWithBaseURL(this.baseURL + "/webxdc_bootstrap324567869.html", bootstrapHtml, "text/html", "UTF-8", null);
-    } catch (IOException e) {
-      e.printStackTrace();
-    }
+    webView.loadUrl(this.baseURL + "/webxdc_bootstrap324567869.html");
```

Comment:
"/webxdc_bootstrap324567869.html" could be a constant to avoid typos from breaking things (OTOH, the typo would break all non-malicious webxdcs, so no way it goes unnoticed)

Reply:
i think, it is good enough :)